server/src/main/java/org/apache/iotdb/db/http/router/Router.java

```diff
@@ -0,0 +1,109 @@
+import org.apache.iotdb.db.http.handler.GetTimeSeriesHandler;
+import org.apache.iotdb.db.http.handler.InsertHandler;
+import org.apache.iotdb.db.http.handler.QueryHandler;
+import org.apache.iotdb.db.http.handler.StorageGroupsHandlers;
+import org.apache.iotdb.db.http.handler.TimeSeriesHandler;
+import org.apache.iotdb.db.http.handler.UsersHandler;
+import org.apache.iotdb.db.http.utils.URIUtils;
+import org.apache.iotdb.tsfile.exception.filter.QueryFilterOptimizationException;
+import org.apache.thrift.TException;
+
+/**
+ * Router that contains information about both route matching and return json.
+ */
+public class Router {
+
+  /**
+   * matching url and return JSON
+   * @param method http method
+   * @param uri uri will be matched
+   * @param json request JSON Object
+   * @return JSON object, may be a JSONArray or JSONObject
+   */
+  public JsonElement route(HttpMethod method, String uri, JsonElement json)
+      throws AuthException, MetadataException, QueryProcessException
+      , StorageEngineException, UnsupportedHttpMethod, SQLException, InterruptedException, QueryFilterOptimizationException, IOException, TException {
+    QueryStringDecoder decoder = new QueryStringDecoder(uri);
+    uri = URIUtils.removeParameter(uri);
+    switch (uri) {
+      case HttpConstant.ROUTING_STORAGE_GROUPS:
+        StorageGroupsHandlers storageGroupsHandlers = new StorageGroupsHandlers();
+        return storageGroupsHandlers.handle(method, json);
+      case HttpConstant.ROUTING_TIME_SERIES:
+        TimeSeriesHandler timeSeriesHandler = new TimeSeriesHandler();
+        return timeSeriesHandler.handle(method, json);
+      case HttpConstant.ROUTING_USER_LOGIN:
+        if (UsersHandler.userLogin(decoder.parameters())) {
+          JsonObject result = new JsonObject();
+          result.addProperty(HttpConstant.RESULT, HttpConstant.SUCCESSFUL_OPERATION);
+          return result;
+        } else {
+          throw new AuthException(String.format("%s can't log in", UsersHandler.getUsername()));
+        }
+      case HttpConstant.ROUTING_USER_LOGOUT:
+        if (UsersHandler.userLogout(decoder.parameters())) {
+          JsonObject result = new JsonObject();
+          result.addProperty(HttpConstant.RESULT, HttpConstant.SUCCESSFUL_OPERATION);
+          return result;
+        } else {
+          throw new AuthException(String.format("%s can't log out", UsersHandler.getUsername()));
+        }
+      case HttpConstant.ROUTING_QUERY:
+        QueryHandler queryHandler = new QueryHandler();
+        return queryHandler.handle(json.getAsJsonObject());
+      case HttpConstant.ROUTING_INSERT:
+        InsertHandler insertHandler = new InsertHandler();
+        return insertHandler.handle(json.getAsJsonArray());
+      case HttpConstant.ROUTING_STORAGE_GROUPS_DELETE:
+        DeleteStorageGroupsHandler deleteStorageGroupsHandler = new DeleteStorageGroupsHandler();
+        return deleteStorageGroupsHandler.handle(json.getAsJsonArray());
```

Comment:
Same as the comment in StorageGroupHandler. Provide one more interface for JsonObject please.